src/routes/files/handlers.js

```diff
@@ -40,8 +50,65 @@ const altFileTypes = [
   },
 ];
 
-export const deleteHandler = async (req, res) => {
-  const { reportId, fileId } = req.params;
+const deleteHandler = async (req, res) => {
+  const {
+    reportId,
+    reportObjectiveId,
+    objectiveId,
+    objectiveTempleteId,
+    fileId,
+  } = req.params;
+  const user = await userById(req.session.userId);
+  const report = await activityReportById(reportId);
+  const authorization = new ActivityReportPolicy(user, report);
+
+  if (!authorization.canUpdate()) {
+    res.sendStatus(403);
+    return;
+  }
+  try {
+    let file = await getFileById(fileId);
+    if (reportId
+      && reportId in file.reportFiles.map((r) => r.activityReportId)) {
+      const rf = file.reportFiles.find((r) => r.reportId === reportId);
+      deleteActivityReportFile(rf.id);
+    } else if (reportObjectiveId
+      && reportObjectiveId in file.reportObjectiveFiles.map((aro) => aro.reportObjectiveId)) {
+      const rof = file.reportObjectiveFiles.find((r) => r.reportObjectiveId === reportObjectiveId);
+      deleteActivityReportObjectiveFile(rof.id);
+    } else if (objectiveId
+      && objectiveId in file.objectiveFiles.map((r) => r.objectiveId)) {
+      const of = file.objectiveFiles.find((r) => r.objectiveId === objectiveId);
+      deleteObjectiveFile(of.id);
+    } else if (objectiveTempleteId
+      && objectiveTempleteId in file.objectiveTemplateFiles.map((r) => r.objectiveTempleteId)) {
+      const otf = file.objectiveTemplateFiles
+        .find((r) => r.objectiveTempleteId === objectiveTempleteId);
+      deleteObjectiveTemplateFile(otf.id);
+    }
+    file = await getFileById(fileId);
```

Comment:
If you want to reload the file from the DB you should be able to just call `await file.reload()` (granted I'm not sure if it properly reloads any associations)